### ISSUE TITLE
Problem Tailing Logs

### ISSUE THREAD

**Issue by kjschulz** [dokku/dokku-scheduler-kubernetes#31]
### Able to reproduce consistently

I'm using **dokku** version `20.4` and **dokku-sechedule-kubernetes** version `17.2`.

### Steps to Reproduce

1. Try running `dokku logs <APP_NAME> --tail`
2. Try running shorthand version:  `dokku logs <APP_NAME> --t`

#### Actual Results

Both options yield the same result.  The output is the default non-tail output as if I had run `dokku logs <APP_NAME>`.

#### Expected Results

I expected to see a tailed stream of error logs in real-time like they do on my other dokku servers.

### `dokku report` output

```
-----> uname: Linux c1-primary 4.15.0-66-generic #75-Ubuntu SMP Tue Oct 1 05:24:09 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
-----> memory: 
                     total        used        free      shared  buff/cache   available
       Mem:           1993         288         368           0        1336        1597
       Swap:             0           0           0
-----> docker version: 
       Client: Docker Engine - Community
        Version:           19.03.8
        API version:       1.40
        Go version:        go1.12.17
        Git commit:        afacb8b7f0
        Built:             Wed Mar 11 01:25:46 2020
        OS/Arch:           linux/amd64
        Experimental:      false
       
       Server: Docker Engine - Community
        Engine:
         Version:          19.03.8
         API version:      1.40 (minimum version 1.12)
         Go version:       go1.12.17
         Git commit:       afacb8b7f0
         Built:            Wed Mar 11 01:24:19 2020
         OS/Arch:          linux/amd64
         Experimental:     false
        containerd:
         Version:          1.2.13
         GitCommit:        7ad184331fa3e55e52b890ea95e65ba581ae3429
        runc:
         Version:          1.0.0-rc10
         GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
        docker-init:
         Version:          0.18.0
         GitCommit:        fec3683
-----> docker daemon info: 
WARNING: No swap limit support
       Client:
        Debug Mode: true
       
       Server:
        Containers: 24
         Running: 0
         Paused: 0
         Stopped: 24
        Images: 29
        Server Version: 19.03.8
        Storage Driver: overlay2
         Backing Filesystem: <unknown>
         Supports d_type: true
         Native Overlay Diff: true
        Logging Driver: json-file
        Cgroup Driver: cgroupfs
        Plugins:
         Volume: local
         Network: bridge host ipvlan macvlan null overlay
         Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
        Swarm: inactive
        Runtimes: runc
        Default Runtime: runc
        Init Binary: docker-init
        containerd version: 7ad184331fa3e55e52b890ea95e65ba581ae3429
        runc version: dc9208a3303feef5b3839f4323d9beb36df0a9dd
        init version: fec3683
        Security Options:
         apparmor
         seccomp
          Profile: default
        Kernel Version: 4.15.0-66-generic
        Operating System: Ubuntu 18.04.3 LTS
        OSType: linux
        Architecture: x86_64
        CPUs: 1
        Total Memory: 1.947GiB
        Name: c1-primary
        ID: YC32:7M7Y:KSRW:DA6S:G73T:ZZ2S:NVZG:QPLI:3WMI:AYBK:HEFI:5VDA
        Docker Root Dir: /var/lib/docker
        Debug Mode: false
        Registry: https://index.docker.io/v1/
        Labels:
        Experimental: false
        Insecure Registries:
         127.0.0.0/8
        Live Restore Enabled: false
       
-----> sigil version: 0.5.0
-----> herokuish version: 
       herokuish: 0.5.12
       buildpacks:
         heroku-buildpack-multi     v1.0.0
         heroku-buildpack-ruby      v215
         heroku-buildpack-nodejs    v170
         heroku-buildpack-clojure   v84
         heroku-buildpack-python    v169
         heroku-buildpack-java      v66
         heroku-buildpack-gradle    v31
         heroku-buildpack-scala     v87
         heroku-buildpack-play      v26
         heroku-buildpack-php       v173
         heroku-buildpack-go        v141
         buildpack-nginx            v13
-----> dokku version: dokku version 0.20.4
-----> dokku plugins: 
       plugn: 0.3.2
         00_dokku-standard    0.20.4 enabled    dokku core standard plugin
         20_events            0.20.4 enabled    dokku core events logging plugin
         app-json             0.20.4 enabled    dokku core app-json plugin
         apps                 0.20.4 enabled    dokku core apps plugin
         builder-dockerfile   0.20.4 enabled    dokku core builder-dockerfile plugin
         builder-herokuish    0.20.4 enabled    dokku core builder-herokuish plugin
         buildpacks           0.20.4 enabled    dokku core buildpacks plugin
         certs                0.20.4 enabled    dokku core certificate management plugin
         checks               0.20.4 enabled    dokku core checks plugin
         common               0.20.4 enabled    dokku core common plugin
         config               0.20.4 enabled    dokku core config plugin
         docker-options       0.20.4 enabled    dokku core docker-options plugin
         domains              0.20.4 enabled    dokku core domains plugin
         enter                0.20.4 enabled    dokku core enter plugin
         git                  0.20.4 enabled    dokku core git plugin
         ha-dokku             0.1.0 enabled    ensures all applications are configured for deployment to a remote registry
         logs                 0.20.4 enabled    dokku core logs plugin
         network              0.20.4 enabled    dokku core network plugin
         nginx-vhosts         0.20.4 enabled    dokku core nginx-vhosts plugin
         plugin               0.20.4 enabled    dokku core plugin plugin
         proxy                0.20.4 enabled    dokku core proxy plugin
         ps                   0.20.4 enabled    dokku core ps plugin
         registry             0.5.1 enabled    manages integration with remote docker registries
         repo                 0.20.4 enabled    dokku core repo plugin
         resource             0.20.4 enabled    dokku core resource plugin
         scheduler-docker-local 0.20.4 enabled    dokku core scheduler-docker-local plugin
         scheduler-kubernetes 0.17.2 enabled    dokku scheduler-kubernetes plugin
         shell                0.20.4 enabled    dokku core shell plugin
         ssh-keys             0.20.4 enabled    dokku core ssh-keys plugin
         storage              0.20.4 enabled    dokku core storage plugin
         tags                 0.20.4 enabled    dokku core tags plugin
         tar                  0.20.4 enabled    dokku core tar plugin
         trace                0.20.4 enabled    dokku core trace plugin
```

### How (deb/make/rpm) and where (AWS, VirtualBox, physical, etc.) was Dokku installed?:

Installed on DO via terraform (`tf-dokku-do-kubernetes`)

### Additional information

- Output of failing Dokku commands after running `dokku trace:on`
  (BEWARE: `trace:on` will print environment variables for some commands, be sure you're not exposing any sensitive information when posting issues. You may replace these values with XXXXXX):

```
+ export DOKKU_HOST_ROOT=/home/dokku
+ DOKKU_HOST_ROOT=/home/dokku
+ export DOKKU_DISTRO
++ . /etc/os-release
++ echo ubuntu
+ DOKKU_DISTRO=ubuntu
+ export DOCKER_BIN=docker
+ DOCKER_BIN=docker
+ export DOKKU_IMAGE=gliderlabs/herokuish:latest
+ DOKKU_IMAGE=gliderlabs/herokuish:latest
+ export DOKKU_LIB_ROOT=/var/lib/dokku
+ DOKKU_LIB_ROOT=/var/lib/dokku
+ export PLUGIN_PATH=/var/lib/dokku/plugins
+ PLUGIN_PATH=/var/lib/dokku/plugins
+ export PLUGIN_AVAILABLE_PATH=/var/lib/dokku/plugins/available
+ PLUGIN_AVAILABLE_PATH=/var/lib/dokku/plugins/available
+ export PLUGIN_ENABLED_PATH=/var/lib/dokku/plugins/enabled
+ PLUGIN_ENABLED_PATH=/var/lib/dokku/plugins/enabled
+ export PLUGIN_CORE_PATH=/var/lib/dokku/core-plugins
+ PLUGIN_CORE_PATH=/var/lib/dokku/core-plugins
+ export PLUGIN_CORE_AVAILABLE_PATH=/var/lib/dokku/core-plugins/available
+ PLUGIN_CORE_AVAILABLE_PATH=/var/lib/dokku/core-plugins/available
+ export PLUGIN_CORE_ENABLED_PATH=/var/lib/dokku/core-plugins/enabled
+ PLUGIN_CORE_ENABLED_PATH=/var/lib/dokku/core-plugins/enabled
+ export DOKKU_SYSTEM_GROUP=dokku
+ DOKKU_SYSTEM_GROUP=dokku
+ export DOKKU_SYSTEM_USER=dokku
+ DOKKU_SYSTEM_USER=dokku
+ export DOKKU_API_VERSION=1
+ DOKKU_API_VERSION=1
+ export DOKKU_NOT_IMPLEMENTED_EXIT=10
+ DOKKU_NOT_IMPLEMENTED_EXIT=10
+ export DOKKU_VALID_EXIT=0
+ DOKKU_VALID_EXIT=0
+ export DOKKU_PID=6177
+ DOKKU_PID=6177
+ export DOKKU_LOGS_DIR=/var/log/dokku
+ DOKKU_LOGS_DIR=/var/log/dokku
+ export DOKKU_EVENTS_LOGFILE=/var/log/dokku/events.log
+ DOKKU_EVENTS_LOGFILE=/var/log/dokku/events.log
+ export DOKKU_CONTAINER_LABEL=dokku
+ DOKKU_CONTAINER_LABEL=dokku
+ export 'DOKKU_GLOBAL_BUILD_ARGS=--label=org.label-schema.schema-version=1.0 --label=org.label-schema.vendor=dokku --label=dokku'
+ DOKKU_GLOBAL_BUILD_ARGS='--label=org.label-schema.schema-version=1.0 --label=org.label-schema.vendor=dokku --label=dokku'
+ export 'DOKKU_GLOBAL_RUN_ARGS=--label=org.label-schema.schema-version=1.0 --label=org.label-schema.vendor=dokku --label=dokku'
+ DOKKU_GLOBAL_RUN_ARGS='--label=org.label-schema.schema-version=1.0 --label=org.label-schema.vendor=dokku --label=dokku'
+ source /var/lib/dokku/core-plugins/available/common/functions
++ set -eo pipefail
++ [[ -n 1 ]]
++ set -x
+ parse_args logs XXXXXX --tail
+ declare 'desc=top-level cli arg parser'
+ local next_index=1
+ local skip=false
+ args=("$@")
+ local args
+ local flags
+ for arg in "$@"
+ [[ false == \t\r\u\e ]]
+ case "$arg" in
+ [[ false == \t\r\u\e ]]
+ [[ logs == \-\-\a\p\p ]]
+ [[ logs =~ ^--.* ]]
+ next_index=2
+ for arg in "$@"
+ [[ false == \t\r\u\e ]]
+ case "$arg" in
+ [[ false == \t\r\u\e ]]
+ [[ XXXXXX == \-\-\a\p\p ]]
+ [[ XXXXXX =~ ^--.* ]]
+ next_index=3
+ for arg in "$@"
+ [[ false == \t\r\u\e ]]
+ case "$arg" in
+ [[ false == \t\r\u\e ]]
+ [[ --tail == \-\-\a\p\p ]]
+ [[ --tail =~ ^--.* ]]
+ flags=' --tail'
+ next_index=4
+ [[ -z '' ]]
++ sed -e 's/^[[:space:]]*//' -e 's/[[:space:]]*$//'
++ echo -e ' --tail'
+ export DOKKU_GLOBAL_FLAGS=--tail
+ DOKKU_GLOBAL_FLAGS=--tail
+ return 0
+ args=("$@")
+ skip_arg=false
+ [[ logs =~ ^--.* ]]
+ has_tty
+ declare 'desc=return 0 if we have a tty'
++ LC_ALL=C
++ /usr/bin/tty
+ [[ /dev/pts/0 == \n\o\t\ \a\ \t\t\y ]]
+ return 0
++ id -un
+ [[ root != \d\o\k\k\u ]]
+ [[ ! logs =~ plugin:* ]]
+ [[ logs != \s\s\h\-\k\e\y\s\:\a\d\d ]]
+ [[ logs != \s\s\h\-\k\e\y\s\:\r\e\m\o\v\e ]]
++ id -un
+ export SSH_USER=root
+ SSH_USER=root
+ sudo -u dokku -E -H /usr/bin/dokku logs XXXXXX --tail
+ export DOKKU_HOST_ROOT=/home/dokku
+ DOKKU_HOST_ROOT=/home/dokku
+ export DOKKU_DISTRO
++ . /etc/os-release
++ echo ubuntu
+ DOKKU_DISTRO=ubuntu
+ export DOCKER_BIN=docker
+ DOCKER_BIN=docker
+ export DOKKU_IMAGE=gliderlabs/herokuish:latest
+ DOKKU_IMAGE=gliderlabs/herokuish:latest
+ export DOKKU_LIB_ROOT=/var/lib/dokku
+ DOKKU_LIB_ROOT=/var/lib/dokku
+ export PLUGIN_PATH=/var/lib/dokku/plugins
+ PLUGIN_PATH=/var/lib/dokku/plugins
+ export PLUGIN_AVAILABLE_PATH=/var/lib/dokku/plugins/available
+ PLUGIN_AVAILABLE_PATH=/var/lib/dokku/plugins/available
+ export PLUGIN_ENABLED_PATH=/var/lib/dokku/plugins/enabled
+ PLUGIN_ENABLED_PATH=/var/lib/dokku/plugins/enabled
+ export PLUGIN_CORE_PATH=/var/lib/dokku/core-plugins
+ PLUGIN_CORE_PATH=/var/lib/dokku/core-plugins
+ export PLUGIN_CORE_AVAILABLE_PATH=/var/lib/dokku/core-plugins/available
+ PLUGIN_CORE_AVAILABLE_PATH=/var/lib/dokku/core-plugins/available
+ export PLUGIN_CORE_ENABLED_PATH=/var/lib/dokku/core-plugins/enabled
+ PLUGIN_CORE_ENABLED_PATH=/var/lib/dokku/core-plugins/enabled
+ export DOKKU_SYSTEM_GROUP=dokku
+ DOKKU_SYSTEM_GROUP=dokku
+ export DOKKU_SYSTEM_USER=dokku
+ DOKKU_SYSTEM_USER=dokku
+ export DOKKU_API_VERSION=1
+ DOKKU_API_VERSION=1
+ export DOKKU_NOT_IMPLEMENTED_EXIT=10
+ DOKKU_NOT_IMPLEMENTED_EXIT=10
+ export DOKKU_VALID_EXIT=0
+ DOKKU_VALID_EXIT=0
+ export DOKKU_PID=6187
+ DOKKU_PID=6187
+ export DOKKU_LOGS_DIR=/var/log/dokku
+ DOKKU_LOGS_DIR=/var/log/dokku
+ export DOKKU_EVENTS_LOGFILE=/var/log/dokku/events.log
+ DOKKU_EVENTS_LOGFILE=/var/log/dokku/events.log
+ export DOKKU_CONTAINER_LABEL=dokku
+ DOKKU_CONTAINER_LABEL=dokku
+ export 'DOKKU_GLOBAL_BUILD_ARGS=--label=org.label-schema.schema-version=1.0 --label=org.label-schema.vendor=dokku --label=dokku'
+ DOKKU_GLOBAL_BUILD_ARGS='--label=org.label-schema.schema-version=1.0 --label=org.label-schema.vendor=dokku --label=dokku'
+ export 'DOKKU_GLOBAL_RUN_ARGS=--label=org.label-schema.schema-version=1.0 --label=org.label-schema.vendor=dokku --label=dokku'
+ DOKKU_GLOBAL_RUN_ARGS='--label=org.label-schema.schema-version=1.0 --label=org.label-schema.vendor=dokku --label=dokku'
+ source /var/lib/dokku/core-plugins/available/common/functions
++ set -eo pipefail
++ [[ -n 1 ]]
++ set -x
+ parse_args logs XXXXXX --tail
+ declare 'desc=top-level cli arg parser'
+ local next_index=1
+ local skip=false
+ args=("$@")
+ local args
+ local flags
+ for arg in "$@"
+ [[ false == \t\r\u\e ]]
+ case "$arg" in
+ [[ false == \t\r\u\e ]]
+ [[ logs == \-\-\a\p\p ]]
+ [[ logs =~ ^--.* ]]
+ next_index=2
+ for arg in "$@"
+ [[ false == \t\r\u\e ]]
+ case "$arg" in
+ [[ false == \t\r\u\e ]]
+ [[ XXXXXX == \-\-\a\p\p ]]
+ [[ XXXXXX =~ ^--.* ]]
+ next_index=3
+ for arg in "$@"
+ [[ false == \t\r\u\e ]]
+ case "$arg" in
+ [[ false == \t\r\u\e ]]
+ [[ --tail == \-\-\a\p\p ]]
+ [[ --tail =~ ^--.* ]]
+ flags=' --tail'
+ next_index=4
+ [[ -z --tail ]]
+ return 0
+ args=("$@")
+ skip_arg=false
+ [[ logs =~ ^--.* ]]
+ has_tty
+ declare 'desc=return 0 if we have a tty'
++ LC_ALL=C
++ /usr/bin/tty
+ [[ /dev/pts/0 == \n\o\t\ \a\ \t\t\y ]]
+ return 0
++ id -un
+ [[ dokku != \d\o\k\k\u ]]
+ [[ logs =~ ^plugin:.* ]]
+ [[ logs == \s\s\h\-\k\e\y\s\:\a\d\d ]]
+ [[ logs == \s\s\h\-\k\e\y\s\:\r\e\m\o\v\e ]]
+ [[ -n '' ]]
+ dokku_auth logs XXXXXX --tail
+ declare 'desc=calls user-auth plugin trigger'
+ export SSH_USER=root
+ SSH_USER=root
+ export SSH_NAME=default
+ SSH_NAME=default
++ wc -l
++ find /var/lib/dokku/plugins/enabled/20_events/user-auth
+ [[ 1 == 1 ]]
+ return 0
+ case "$1" in
+ execute_dokku_cmd logs XXXXXX --tail
+ declare 'desc=executes dokku sub-commands'
+ local PLUGIN_NAME=logs
+ local PLUGIN_CMD=logs
+ local implemented=0
+ local script
+ argv=("$@")
+ local argv
+ case "$PLUGIN_NAME" in
++ readlink -f /var/lib/dokku/plugins/enabled/logs
+ [[ /var/lib/dokku/core-plugins/available/logs == *core-plugins* ]]
+ [[ logs == \l\o\g\s ]]
+ shift 1
+ [[ -n '' ]]
+ set -- logs XXXXXX --tail
+ [[ -x /var/lib/dokku/plugins/enabled/logs/subcommands/default ]]
+ /var/lib/dokku/plugins/enabled/logs/subcommands/default logs XXXXXX --tail
+ source /var/lib/dokku/core-plugins/available/common/functions
++ set -eo pipefail
++ [[ -n 1 ]]
++ set -x
+ source /var/lib/dokku/plugins/available/logs/internal-functions
++ set -eo pipefail
++ [[ -n 1 ]]
++ set -x
+ cmd-logs-default logs XXXXXX --tail
+ declare 'desc=display recent log output'
+ declare cmd=logs
+ [[ logs == \l\o\g\s ]]
+ shift 1
+ declare APP=XXXXXX
+ [[ -z XXXXXX ]]
+ verify_app_name XXXXXX
+ declare 'desc=verify app name format and app existence'
+ local APP=XXXXXX
+ is_valid_app_name XXXXXX
+ declare 'desc=verify app name format'
+ local APP=XXXXXX
+ [[ -z XXXXXX ]]
+ [[ XXXXXX =~ ^[a-z].* ]]
+ [[ ! XXXXXX =~ [A-Z] ]]
+ [[ ! XXXXXX =~ [:] ]]
+ return 0
+ [[ ! -d /home/dokku/XXXXXX ]]
+ return 0
+ is_deployed XXXXXX
+ declare 'desc=return 0 if given app has a running container'
+ local APP=XXXXXX
+ source /var/lib/dokku/plugins/available/config/functions
++ set -eo pipefail
++ [[ -n 1 ]]
++ set -x
++ source /var/lib/dokku/core-plugins/available/common/functions
+++ set -eo pipefail
+++ [[ -n 1 ]]
+++ set -x
++ get_app_scheduler XXXXXX
++ declare 'desc=fetch the scheduler for a given application'
++ declare APP=XXXXXX
++ local DOKKU_APP_SCHEDULER DOKKU_GLOBAL_SCHEDULER DOKKU_SCHEDULER
++ [[ XXXXXX == \-\-\g\l\o\b\a\l ]]
++ source /var/lib/dokku/plugins/available/config/functions
+++ set -eo pipefail
+++ [[ -n 1 ]]
+++ set -x
+++ source /var/lib/dokku/core-plugins/available/common/functions
++++ set -eo pipefail
++++ [[ -n 1 ]]
++++ set -x
++ [[ -n XXXXXX ]]
+++ config_get XXXXXX DOKKU_SCHEDULER
+++ declare 'desc=get value of given config var'
+++ config_sub get XXXXXX DOKKU_SCHEDULER
+++ declare 'desc=executes a config subcommand'
+++ local name=get
+++ shift
+++ /var/lib/dokku/plugins/available/config/subcommands/get config:get XXXXXX DOKKU_SCHEDULER
+++ true
++ DOKKU_APP_SCHEDULER=
+++ config_get --global DOKKU_SCHEDULER
+++ declare 'desc=get value of given config var'
+++ config_sub get --global DOKKU_SCHEDULER
+++ declare 'desc=executes a config subcommand'
+++ local name=get
+++ shift
+++ /var/lib/dokku/plugins/available/config/subcommands/get config:get --global DOKKU_SCHEDULER
++ DOKKU_GLOBAL_SCHEDULER=kubernetes
++ DOKKU_SCHEDULER=kubernetes
++ [[ -z kubernetes ]]
++ echo kubernetes
+ local DOKKU_SCHEDULER=kubernetes
+ plugn trigger scheduler-is-deployed kubernetes XXXXXX
+ source /var/lib/dokku/core-plugins/available/common/functions
++ set -eo pipefail
++ [[ -n 1 ]]
++ set -x
+ [[ ! -n '' ]]
+ trigger-scheduler-docker-local-scheduler-is-deployed kubernetes XXXXXX
+ declare 'desc=checks if an app is deployed'
+ declare trigger=scheduler-is-deployed
+ declare DOKKU_SCHEDULER=kubernetes APP=XXXXXX
+ [[ kubernetes != \d\o\c\k\e\r\-\l\o\c\a\l ]]
+ return
+ source /var/lib/dokku/core-plugins/available/common/functions
++ set -eo pipefail
++ [[ -n 1 ]]
++ set -x
+ source /var/lib/dokku/core-plugins/available/common/property-functions
++ set -eo pipefail
++ [[ -n 1 ]]
++ set -x
+ source /var/lib/dokku/plugins/available/scheduler-kubernetes/internal-functions
++ source /var/lib/dokku/core-plugins/available/common/functions
+++ set -eo pipefail
+++ [[ -n 1 ]]
+++ set -x
++ source /var/lib/dokku/core-plugins/available/common/property-functions
+++ set -eo pipefail
+++ [[ -n 1 ]]
+++ set -x
++ source /var/lib/dokku/plugins/available/config/functions
+++ set -eo pipefail
+++ [[ -n 1 ]]
+++ set -x
+++ source /var/lib/dokku/core-plugins/available/common/functions
++++ set -eo pipefail
++++ [[ -n 1 ]]
++++ set -x
++ source /var/lib/dokku/plugins/available/domains/functions
+++ set -eo pipefail
+++ [[ -n 1 ]]
+++ set -x
+++ source /var/lib/dokku/core-plugins/available/common/functions
++++ set -eo pipefail
++++ [[ -n 1 ]]
++++ set -x
++ set -eo pipefail
++ [[ -n 1 ]]
++ set -x
+ scheduler-kubernetes-scheduler-is-deployed kubernetes XXXXXX
+ declare 'desc=scheduler-kubernetes scheduler-is-deployed plugin trigger'
+ declare 'trigger=scheduler-kubernetes scheduler-is-deployed'
+ declare DOKKU_SCHEDULER=kubernetes APP=XXXXXX
+ local KUBE_OUTPUT
+ [[ kubernetes != \k\u\b\e\r\n\e\t\e\s ]]
+ export KUBECONFIG=/home/dokku/.kube/config
+ KUBECONFIG=/home/dokku/.kube/config
+ export KUBEDOG_KUBE_CONFIG=/home/dokku/.kube/config
+ KUBEDOG_KUBE_CONFIG=/home/dokku/.kube/config
+ KUBE_ARGS=()
++ fn-plugin-property-get scheduler-kubernetes XXXXXX namespace ''
++ declare 'desc=returns the value for a given property'
++ declare PLUGIN=scheduler-kubernetes APP=XXXXXX KEY=namespace DEFAULT=
++ fn-plugin-property-get-default scheduler-kubernetes XXXXXX namespace ''
++ declare 'desc=returns the value for a given property with a specified default value'
++ declare PLUGIN=scheduler-kubernetes APP=XXXXXX KEY=namespace DEFAULT=
++ /var/lib/dokku/core-plugins/available/common/prop get-with-default scheduler-kubernetes XXXXXX namespace ''
+ NAMESPACE=
++ fn-plugin-property-get scheduler-kubernetes XXXXXX namespace default
++ declare 'desc=returns the value for a given property'
++ declare PLUGIN=scheduler-kubernetes APP=XXXXXX KEY=namespace DEFAULT=default
++ fn-plugin-property-get-default scheduler-kubernetes XXXXXX namespace default
++ declare 'desc=returns the value for a given property with a specified default value'
++ declare PLUGIN=scheduler-kubernetes APP=XXXXXX KEY=namespace DEFAULT=default
++ /var/lib/dokku/core-plugins/available/common/prop get-with-default scheduler-kubernetes XXXXXX namespace default
+ NAMESPACE=default
+ KUBE_ARGS+=("--namespace=$NAMESPACE")
+ fn-scheduler-kubernetes-ensure-namespace default
+ declare NAMESPACE=default
+ [[ default = \d\e\f\a\u\l\t ]]
+ return
++ /var/lib/dokku/data/scheduler-kubernetes/kubectl --namespace=default get pods --selector app=XXXXXX -o name
+ KUBE_OUTPUT='pod/XXXXXX-web-85759c6c55-9hpkj
pod/XXXXXX-web-85759c6c55-d6797
pod/XXXXXX-web-85759c6c55-pwmc8'
+ [[ -n pod/XXXXXX-web-85759c6c55-9hpkj
pod/XXXXXX-web-85759c6c55-d6797
pod/XXXXXX-web-85759c6c55-pwmc8 ]]
+ return 0
+ shift 1
++ getopt -o htqn:p: --long help,tail,quiet,num:,ps: -n 'dokku logs' -- --tail
+ local 'TEMP= --tail --'
+ local EXIT_CODE=0
+ [[ 0 != 0 ]]
+ eval set -- ' --tail --'
++ set -- --tail --
+ local NUM=100
+ true
+ case "$1" in
+ local TAIL=true
+ shift
+ true
+ case "$1" in
+ shift
+ break
++ get_app_scheduler XXXXXX
++ declare 'desc=fetch the scheduler for a given application'
++ declare APP=XXXXXX
++ local DOKKU_APP_SCHEDULER DOKKU_GLOBAL_SCHEDULER DOKKU_SCHEDULER
++ [[ XXXXXX == \-\-\g\l\o\b\a\l ]]
++ source /var/lib/dokku/plugins/available/config/functions
+++ set -eo pipefail
+++ [[ -n 1 ]]
+++ set -x
+++ source /var/lib/dokku/core-plugins/available/common/functions
++++ set -eo pipefail
++++ [[ -n 1 ]]
++++ set -x
++ [[ -n XXXXXX ]]
+++ config_get XXXXXX DOKKU_SCHEDULER
+++ declare 'desc=get value of given config var'
+++ config_sub get XXXXXX DOKKU_SCHEDULER
+++ declare 'desc=executes a config subcommand'
+++ local name=get
+++ shift
+++ /var/lib/dokku/plugins/available/config/subcommands/get config:get XXXXXX DOKKU_SCHEDULER
+++ true
++ DOKKU_APP_SCHEDULER=
+++ config_get --global DOKKU_SCHEDULER
+++ declare 'desc=get value of given config var'
+++ config_sub get --global DOKKU_SCHEDULER
+++ declare 'desc=executes a config subcommand'
+++ local name=get
+++ shift
+++ /var/lib/dokku/plugins/available/config/subcommands/get config:get --global DOKKU_SCHEDULER
++ DOKKU_GLOBAL_SCHEDULER=kubernetes
++ DOKKU_SCHEDULER=kubernetes
++ [[ -z kubernetes ]]
++ echo kubernetes
+ DOKKU_SCHEDULER=kubernetes
+ plugn trigger scheduler-logs kubernetes XXXXXX '' true '' 100
+ source /var/lib/dokku/core-plugins/available/common/functions
++ set -eo pipefail
++ [[ -n 1 ]]
++ set -x
+ [[ ! -n '' ]]
+ source /var/lib/dokku/core-plugins/available/common/functions
++ set -eo pipefail
++ [[ -n 1 ]]
++ set -x
+ trigger-scheduler-docker-local-scheduler-logs kubernetes XXXXXX '' true '' 100
+ declare 'desc=scheduler-docker-local scheduler-logs plugin trigger'
+ declare trigger=scheduler-logs
+ declare DOKKU_SCHEDULER=kubernetes APP=XXXXXX PROCESS_TYPE= TAIL=true PRETTY_PRINT= NUM=100
+ local DOKKU_LOGS_ARGS=
+ [[ kubernetes != \d\o\c\k\e\r\-\l\o\c\a\l ]]
+ return
+ source /var/lib/dokku/core-plugins/available/common/functions
++ set -eo pipefail
++ [[ -n 1 ]]
++ set -x
+ source /var/lib/dokku/core-plugins/available/common/property-functions
++ set -eo pipefail
++ [[ -n 1 ]]
++ set -x
+ source /var/lib/dokku/plugins/available/scheduler-kubernetes/internal-functions
++ source /var/lib/dokku/core-plugins/available/common/functions
+++ set -eo pipefail
+++ [[ -n 1 ]]
+++ set -x
++ source /var/lib/dokku/core-plugins/available/common/property-functions
+++ set -eo pipefail
+++ [[ -n 1 ]]
+++ set -x
++ source /var/lib/dokku/plugins/available/config/functions
+++ set -eo pipefail
+++ [[ -n 1 ]]
+++ set -x
+++ source /var/lib/dokku/core-plugins/available/common/functions
++++ set -eo pipefail
++++ [[ -n 1 ]]
++++ set -x
++ source /var/lib/dokku/plugins/available/domains/functions
+++ set -eo pipefail
+++ [[ -n 1 ]]
+++ set -x
+++ source /var/lib/dokku/core-plugins/available/common/functions
++++ set -eo pipefail
++++ [[ -n 1 ]]
++++ set -x
++ set -eo pipefail
++ [[ -n 1 ]]
++ set -x
+ scheduler-kubernetes-scheduler-logs kubernetes XXXXXX '' true '' 100
+ declare 'desc=scheduler-kubernetes scheduler-logs plugin trigger'
+ declare 'trigger=scheduler-kubernetes scheduler-logs'
+ declare DOKKU_SCHEDULER=kubernetes APP=XXXXXX PROCESS_TYPE= TAIL=true PRETTY_PRINT= NUM=100
+ local KUBE_ARGS NAMESPACE
+ [[ kubernetes != \k\u\b\e\r\n\e\t\e\s ]]
+ local LABEL=app=XXXXXX
+ [[ -n '' ]]
+ export KUBECONFIG=/home/dokku/.kube/config
+ KUBECONFIG=/home/dokku/.kube/config
+ export KUBEDOG_KUBE_CONFIG=/home/dokku/.kube/config
+ KUBEDOG_KUBE_CONFIG=/home/dokku/.kube/config
+ KUBE_ARGS=()
++ fn-plugin-property-get scheduler-kubernetes XXXXXX namespace default
++ declare 'desc=returns the value for a given property'
++ declare PLUGIN=scheduler-kubernetes APP=XXXXXX KEY=namespace DEFAULT=default
++ fn-plugin-property-get-default scheduler-kubernetes XXXXXX namespace default
++ declare 'desc=returns the value for a given property with a specified default value'
++ declare PLUGIN=scheduler-kubernetes APP=XXXXXX KEY=namespace DEFAULT=default
++ /var/lib/dokku/core-plugins/available/common/prop get-with-default scheduler-kubernetes XXXXXX namespace default
+ NAMESPACE=default
+ KUBE_ARGS+=("--namespace=$NAMESPACE")
+ fn-scheduler-kubernetes-ensure-namespace default
+ declare NAMESPACE=default
+ [[ default = \d\e\f\a\u\l\t ]]
+ return
+ local 'LOGS_ARGS=--tail 100'
+ [[ true == \t\r\u\e ]]
+ LOGS_ARG+=' --follow '
+ /var/lib/dokku/data/scheduler-kubernetes/kubectl --namespace=default logs -l app=XXXXXX
    at /app/node_modules/express/lib/router/index.js:281:22
    at Function.process_params (/app/node_modules/express/lib/router/index.js:335:12)
    at next (/app/node_modules/express/lib/router/index.js:275:10)
    at SendStream.error (/app/node_modules/serve-static/index.js:121:7)
    at SendStream.emit (events.js:314:20)
    at SendStream.EventEmitter.emit (domain.js:483:12)
    at SendStream.error (/app/node_modules/send/index.js:270:17)
    at SendStream.onStatError (/app/node_modules/send/index.js:421:12)
    at next (/app/node_modules/send/index.js:736:16)
    at onstat (/app/node_modules/send/index.js:725:14)
    at /app/node_modules/express/lib/router/index.js:281:22
    at Function.process_params (/app/node_modules/express/lib/router/index.js:335:12)
    at next (/app/node_modules/express/lib/router/index.js:275:10)
    at SendStream.error (/app/node_modules/serve-static/index.js:121:7)
    at SendStream.emit (events.js:314:20)
    at SendStream.EventEmitter.emit (domain.js:483:12)
    at SendStream.error (/app/node_modules/send/index.js:270:17)
    at SendStream.onStatError (/app/node_modules/send/index.js:421:12)
    at next (/app/node_modules/send/index.js:736:16)
    at onstat (/app/node_modules/send/index.js:725:14)
    at /app/node_modules/express/lib/router/index.js:281:22
    at Function.process_params (/app/node_modules/express/lib/router/index.js:335:12)
    at next (/app/node_modules/express/lib/router/index.js:275:10)
    at SendStream.error (/app/node_modules/serve-static/index.js:121:7)
    at SendStream.emit (events.js:314:20)
    at SendStream.EventEmitter.emit (domain.js:483:12)
    at SendStream.error (/app/node_modules/send/index.js:270:17)
    at SendStream.onStatError (/app/node_modules/send/index.js:421:12)
    at next (/app/node_modules/send/index.js:736:16)
    at onstat (/app/node_modules/send/index.js:725:14)
+ implemented=1
+ [[ 1 -eq 0 ]]
+ [[ 1 -eq 0 ]]
+ exit 0
```

**Comment by BRONSOLO** [comment]
@josegonzalez I believe the fix introduced here will need the flag(s) to be specified outside of double quotes. Here is the error I'm seeing using the changes of https://github.com/dokku/dokku-scheduler-kubernetes/commit/6b0bfd40229d15287dd60a792f8e3c1e41294409: 

![Screenshot from 2021-01-20 11-02-39](https://user-images.githubusercontent.com/1909107/105203867-978b6a00-5b11-11eb-9bab-d7c47e41bd32.png)

Thanks!

**Comment by josegonzalez** [comment]
Fixed in 0.17.4.
